Replace Hash Algorithm in HashMap/Set with FxHashMap/Set for simple types

Narfinger opened the pull request:
FxHash is faster than FnvHash and SipHash for  simple types up to at least 64 bytes. The cryptographic guarantees are not needed for any types changed here because they are simple ids.
This changes the types in script and net crates.
In a future PR we will change the remaining Fnv to be also Fx unless there is a reason to keep them as Fnv.

Testing: Should not change functionality but unit test and wpt will find it.

mrobinson commented:
It would be good to measure Speedometer before and after this change.

Narfinger commented:
There might be a slight improvement in memory usage. A RandomState uses 2 u64 for state and FxHash does not have any state. But I don't know if it is measurable.

sagudev reviewed (changes requested):
Instead of unmaintained fxhash we need to use rustc-hash. Related: https://github.com/servo/servo/pull/39168

Narfinger commented:
Yes I will change that on Monday.

Narfinger committed:
This PR replaces the HashMaps and HashSets in components/script and
components/net with FxHash*.
FxHash is one of the fastest non-cryptographic hash functions for small ranges.
It is faster than sip-hash (the standard Hashmap) and in testing faster
than Fnv.

Signed-off-by: Narfinger <Narfinger@users.noreply.github.com>

Narfinger commented:
Ok the change should be done now.

sagudev commented on components/net/image_cache.rs:
```diff
@@ -185,11 +186,11 @@ type ImageKey = (ServoUrl, ImmutableOrigin, Option<CorsSettings>);
 
     // Get a load key from its url and requesting origin. Used ony when starting and
     // finishing a load or when adding a new listener.
-    url_to_load_key: HashMap<ImageKey, LoadKey>,
```
Also because it uses user input (url), I am not sure we should use fx here.

Narfinger commented on components/shared/constellation/Cargo.toml:
```diff
@@ -24,6 +24,7 @@ embedder_traits = { workspace = true }
 euclid = { workspace = true }
 fonts_traits = { workspace = true }
 fnv = { workspace = true }
```
Currently we still need it. I plan to address that in a later PR.

sagudev reviewed (approved):
LGTM, but needs rebase.
